### PR TITLE
type conversion and other stuff

### PR DESCRIPTION
**Pull request by readevalprint**
Added type conversion from the xml to json.
Fixed debug port for odk-collect-importer.

### CODE REVIEW

File: gather2-odk-importer/api/views.py [hunk 1]
```diff
@@ -31,7 +66,7 @@ def form_list(request):
 @authentication_classes([BasicAuthentication])
 @permission_classes([IsAuthenticated])
 def download_xform(request, pk):
-    xform = XForm.objects.get(pk=pk, username=request.user.username)
+    xform = get_object_or_404(XForm, pk=pk, username=request.user.username)
     print(xform)
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Debug output?

File: gather2-odk-importer/api/views.py [hunk 2]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
+                for d in dict_generator(value, [key] + pre):
+                    yield d
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for v in value:
+                    for d in dict_generator(v, [key] + pre):
+                        yield d
+            else:
+                yield pre + [key, value]
+    else:
+        yield indict
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: I keep reading this variable as "indict", but I guess that's not what you mean :)

File: gather2-odk-importer/api/views.py [hunk 3]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
-    title = list(d.keys())[0]  # TODO: Make this better
-    xform = XForm.objects.get(title=title)  # TODO: Make this better
-    print(xform.gather_core_url)
-
-    1 / 0
-    return Response(status=status.HTTP_201_CREATED)
+    if request.method == 'POST':
+        xml = request.FILES['xml_submission_file'].read()
+        d = xmltodict.parse(xml)
+        print(json.dumps(d, sort_keys=True, indent=4))
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Debug output?

File: gather2-odk-importer/api/views.py [hunk 4]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
+
+        title = list(d.items())[0][1]['@id']
+        xform = XForm.objects.get(title=title)
+        print('Gather2 survey urls:', xform.gather_core_url)
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Debug output?

File: gather2-odk-importer/api/views.py [hunk 5]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
+        coerce_dict = {}
+        for n in re.findall(r"<bind.*/>", xform.xml_data):
+            coerce_dict[re.findall(r'nodeset="([^"]*)"', n)[0]] = re.findall(r'type="([^"]*)"', n)[0]
+        walk(d, None, coerce_dict)  # modifies inplace
+        r = requests.post(xform.gather_core_url, json={'data':d, 'survey': 1})
+
+        print(r.text)
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Debug output?

File: gather2-odk-importer/api/views.py [hunk 6]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
+        import ipdb; ipdb.set_trace() # BREAKPOINT
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: &lt;cough&gt;

File: gather2-odk-importer/api/views.py [hunk 7]
```diff
@@ -48,16 +83,44 @@ def xform_manifest(request, id_string):
...
+        1/0
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Is this some secret Django thing I don't know about?

File: gather2-odk-importer/api/views.py [hunk 8]
```diff
@@ -1,11 +1,46 @@
+from functools import reduce
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Are you using `reduce`?

File: gather2-odk-importer/api/views.py [hunk 9]
```diff
@@ -1,11 +1,46 @@
...
+from dateutil import parser
+import geojson
+import re
+import operator
```
Comment (adamvert) [inline gather2-odk-importer/api/views.py]: Ditto `operator`?

File: gather2-core/.coveragerc [hunk 10]
```diff
@@ -4,7 +4,7 @@ omit = /code/core/apps.py
 source = /code/core
 
 [paths]
-source = /code/core
+source = /code/gather2-core/core
 omit = /code/core/apps.py
```
Comment (readevalprint) [inline gather2-core/.coveragerc]: yup!

File: gather2-odk-importer/api/views.py (hunk 7 above)
Comment (readevalprint) [inline gather2-odk-importer/api/views.py]: hahah no it was debug stuff I forgot to remove

**Pull request merged**